src/interchange/avro.rs

```diff
@@ -612,13 +613,27 @@ impl Encoder {
                         .0;
                     let avro_val = Self::data_to_avro(node.step(&before.schema), &row.unpack())?;
                     // Add wrapper Record with before and after RecordFields
-                    let wrapped_avro_val = Value::Record(vec![
-                        (
-                            "before".into(),
-                            Value::Union(null_idx, Box::from(Value::Null)),
-                        ),
-                        ("after".into(), avro_val),
-                    ]);
+                    let wrapped_avro_val = if diff == -1 {
+                        Value::Record(vec![
+                            ("before".into(), avro_val),
+                            (
+                                "after".into(),
+                                Value::Union(null_idx, Box::from(Value::Null)),
```

Comment:
`null_idx` needs to be recomputed here for after, as well.

Reply:
this was an oversight :( and I guess it didn't fail in my testing because in our generated debezium schemas NULL is the second enum variant in both before and after

Reply:
Super convenient to assume that they're the same, though, which inspired the approach in #2465.